tests/unit/adapters/osf-adapter-test.js

```diff
@@ -1,12 +1,401 @@
+    });
+    Ember.run.end();
+
+    var mockAjax = this.stub(adapter, 'ajax', () => {
+        return new Ember.RSVP.Promise(resolve => resolve({}));
+    });
+    adapter._doRelatedRequest(
+        store,
+        node._internalModel.createSnapshot(),
+        children.map(c => c._internalModel.createSnapshot()),
+        'children',
+        'https://api.osf.io/v2/nodes/foobar/children/',
+        'POST'
+    );
+    var data = mockAjax.args[0][2].data.data;
+    assert.equal(data[0].attributes.title, children[0].get('title'));
+    assert.equal(data[1].attributes.title, children[1].get('title'));
+    assert.equal(data[2].attributes.title, children[2].get('title'));
 });
 
-// Replace this with your real tests.
-test('it exists', function(assert) {
-  let adapter = this.subject();
-  assert.ok(adapter);
+test('#_doRelatedRequest with single snapshot', function(assert) {
+    let adapter = this.subject();
+
+    this.inject.service('store');
+    let store = this.store;
+
+    let node = FactoryGuy.make('node');
+    Ember.run.begin();
+    let child = store.createRecord(
+        'node',
+        store.normalize('node', FactoryGuy.build('node').data).data.attributes
+    );
+    Ember.run.end();
+
+    var mockAjax = this.stub(adapter, 'ajax', () => {
+        return new Ember.RSVP.Promise(resolve => resolve({}));
+    });
+    adapter._doRelatedRequest(
+        store,
+        node._internalModel.createSnapshot(),
+        child._internalModel.createSnapshot(),
+        'children',
+        'https://api.osf.io/v2/nodes/foobar/children/',
+        'POST'
+    );
+    var data = mockAjax.args[0][2].data.data;
+    assert.equal(data.attributes.title, child.get('title'));
+});
+
+test('#_handleRelatedRequest makes correct calls for each change argument', function(assert) {
+    let adapter = this.subject();
+
+    this.inject.service('store');
+    let store = this.store;
+
+    let node = FactoryGuy.make('node');
+    let changes = {
+        delete: FactoryGuy.makeList('node', 2),
+        remove: FactoryGuy.makeList('node', 2),
+        update: FactoryGuy.makeList('node', 2),
+        add: FactoryGuy.makeList('node', 2),
+        create: FactoryGuy.makeList('node', 2)
+    };
+    node.set('_dirtyRelationships', {
+        children: changes
+    });
+
+    for (let verb of['delete', 'remove', 'update', 'add', 'create']) {
+        let NodeAdapter = store.adapterFor('node');
+        let relatedStub = this.stub(NodeAdapter, `_${verb}Related`);
+        adapter._handleRelatedRequest(
+            store,
+            store.modelFor('node'),
+            node._internalModel.createSnapshot(),
+            'children',
+            verb
+        );
+        assert.ok(relatedStub.called);
+        assert.deepEqual(
+            relatedStub.args[0][2].map(s => s.id),
+            changes[verb].map(r => r.get('id'))
+        );
+    }
+});
+
+test('#_handleRelatedRequest checks if relationship supports bulk', function(assert) {
```

Comment:
Should this also check that the correct ajaxOptions (`ret.contentType`) are being set?

It's possible that deep bulk support is slated for a separate PR, in which case exhaustive bulk testing could wait.